internal/resources/cluster/integration/default_integration_client.go

```diff
@@ -0,0 +1,32 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package integration
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/integration"
+)
+
+var ErrMissingIntegrationClientService = errors.New("missing integration client service")
```

Comment:
what is the use of this file ? sorry I was not able to connect the dots. Could you please explain ?

We have the cluster integration http client defined in the clients package right

Reply:
Short answer is that it allows using the interface rather than the concrete type, making it easier to switch in a fake for unit tests.  I'll document this a bit better before marking this PR ready again.